### PR TITLE
feature/CLS2-460-add-adviser-field-to-task-form

### PR DESCRIPTION
**Pull request by chopkinsmade**
## Description of change

- Add a new form field to the task form to allow assigning to other advisers

## Test instructions

Either add a new task, or edit an existing task. You will now see the form field to allow you to either assign the task to yourself or another adviser

### Before

image

### After

image

## Checklist

[//]: # 'When submitting a PR make sure the code review guidelines have been satisfied.
https://github.com/uktrade/data-hub-frontend/blob/main/docs/Code%20review%20guidelines.md'

- [ ] Has the branch been rebased to main?
- [ ] Automated tests (Any of the following when applicable: Unit, Functional or End-to-End)
- [ ] Manual compatibility testing (Browsers: Chrome, Firefox, Edge, Safari)

### CODE REVIEW

**Review by cgsunkel — APPROVED** [review]

File: src/client/modules/Tasks/TaskForm/constants.js [hunk 1]
```diff
@@ -0,0 +1,6 @@
+export const OPTION_ME = 'me'
+export const OPTION_SOMEONE_ELSE = 'someoneElse'
+export const OPTIONS_ASSIGNED_TO = [
+  { label: 'Me', value: OPTION_ME },
+  { label: 'Someone else', value: OPTION_SOMEONE_ELSE },
+]
```
Comment (paulgain) [inline src/client/modules/Tasks/TaskForm/constants.js:6]: Consider grouping these as it's tidier and when importing you only have a single import. 
```js
export const OPTIONS = {
  ME: 'me',
  SOMEONE_ELSE: 'someoneElse',
  ASSIGNED_TO: [
    { label: 'Me', value: OPTION_ME },
    { label: 'Someone else', value: OPTION_SOMEONE_ELSE }
  ],
}

import { OPTIONS } from './constants'
...
OPTIONS.ME // 'me'
```

**Review by DeanElliott96 — APPROVED** [review]

File: src/client/modules/Investments/Projects/Tasks/transformers.js [hunk 2]
```diff
@@ -15,7 +19,13 @@ export const transformFormValuesForAPI = (
   task: transformTaskFormValuesForAPI(formValues, currentAdviserId),
 })
 
-export const transformAPIValuesForForm = (task) => ({
+const transformAdvisor = (advisers, currentAdviserId) =>
+  advisers.length === 1 &&
+  advisers.filter((a) => a.id === currentAdviserId).length == 1
+    ? OPTION_ME
+    : OPTION_SOMEONE_ELSE
```
Comment (paulgain) [inline src/client/modules/Investments/Projects/Tasks/transformers.js:26]: The `===` is unnecessary when using find:
```js
const transformAdvisor = (advisers, currentAdviserId) =>
  advisers.find((adviser) => adviser.id === currentAdviserId)
    ? OPTION_ME
    : OPTION_SOMEONE_ELSE

```

**Review by paulgain — COMMENTED** [review]
Looks good, just a couple of comments.

File: src/client/modules/Investments/Projects/Tasks/transformers.js (hunk 2 above)
Comment (chopkinsmade) [reply src/client/modules/Investments/Projects/Tasks/transformers.js:26]: I needed to keep the `advisers.length === 1 &&` as the current advisor could be one of many, but changed to ` advisers.filter((a) => a.id === currentAdviserId).length == 1` to `advisers.find((adviser) => adviser.id === currentAdviserId)`

File: src/client/modules/Tasks/TaskForm/constants.js (hunk 1 above)
Comment (paulgain) [reply src/client/modules/Tasks/TaskForm/constants.js]: @chopkinsmade apologies, the above should have been:
```js
export const OPTIONS = {
  ME: 'me',
  SOMEONE_ELSE: 'someoneElse',
  ASSIGNED_TO: [
    { label: 'Me', value: OPTION.ME }, // Swapped the `_` for a `.`
    { label: 'Someone else', value: OPTION.SOMEONE_ELSE } // Swapped the `_` for a `.`
  ],
}
```
There's no need for the two constants you added later.

Comment (chopkinsmade) [reply src/client/modules/Tasks/TaskForm/constants.js]: Using the 2nd approach you get an undefined error on app start, I'm guessing something to do with access a property from within itself. Also tried with `{ label: 'Me', value: this.ME }` and got same issue

```
constants.js:5 Uncaught TypeError: Cannot read properties of undefined (reading 'ME')
    at ./src/client/modules/Tasks/TaskForm/constants.js (constants.js:5:32)
```

Comment (paulgain) [reply src/client/modules/Tasks/TaskForm/constants.js]: Fair enough!

**Pull request merged**